integrations/access/accesslist/app.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accesslist
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	alclient "github.com/gravitational/teleport/api/client/accesslist"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/access/common/recipient"
+	"github.com/gravitational/teleport/integrations/access/common/teleport"
+	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+const (
+	// oneWeek is the number of hours in a week.
+	oneWeek = 24 * time.Hour * 7
+)
+
+type App[M MessagingBot] struct {
+	pluginName  string
+	apiClient   teleport.Client
+	accessLists services.AccessListsGetter
+	pluginData  *pd.CompareAndSwap[pd.AccessListNotificationData]
+	bot         M
+	job         lib.ServiceJob
+	clock       clockwork.Clock
+}
+
+func NewApp[M MessagingBot]() *App[M] {
+	app := &App[M]{}
+	app.job = lib.NewServiceJob(app.run)
+	return app
+}
+
+func (a *App[M]) Init(baseApp *common.BaseApp[M]) error {
+	a.pluginName = baseApp.PluginName
+	a.apiClient = baseApp.APIClient
+	a.accessLists = accessListClient(a.apiClient)
+	if a.accessLists == nil {
+		return trace.BadParameter("api client does not contain an access list client")
+	}
+	a.pluginData = pd.NewCAS(
+		a.apiClient,
+		a.pluginName,
+		types.KindAccessList,
+		pd.EncodeAccessListNotificationData,
+		pd.DecodeAccessListNotificationData,
+	)
+	a.bot = baseApp.Bot
+
+	if a.clock == nil {
+		a.clock = clockwork.NewRealClock()
+	}
+
+	return nil
+}
+
+func (a *App[_]) Start(ctx context.Context, process *lib.Process) (err error) {
```

Comment:
```suggestion
func (a *App[_]) Start(ctx context.Context, process *lib.Process) error {
```

Any particular reason you used a named return?

Also, why do you need to take a context if you never use it?

Reply:
Artifact of going over this a fair bit, I'll remove it and the error since it's not used either.